pkg/kubecost/coverage.go

```diff
@@ -0,0 +1,108 @@
+package kubecost
+
+import (
+	"time"
+
+	"github.com/opencost/opencost/pkg/filter"
+)
+
+// Coverage This is a placeholder struct which can be replaced by a more specific implementation later
+type Coverage struct {
+	Window   Window    `json:"window"`
+	Type     string    `json:"type"`
+	Count    int       `json:"count"`
+	Updated  time.Time `json:"updated"`
+	Errors   []string  `json:"errors"`
+	Warnings []string  `json:"warnings"`
+}
+
+func (c *Coverage) GetWindow() Window {
+	return c.Window
+}
+
+func (c *Coverage) Key() string {
+	return c.Type
+}
+
+func (c *Coverage) IsEmpty() bool {
+	return c.Type == "" && c.Count == 0 && len(c.Errors) == 0 && len(c.Warnings) == 0 && c.Updated == time.Time{}
+}
+
+func (c *Coverage) Clone() *Coverage {
+	return &Coverage{
+		Window:   c.Window.Clone(),
+		Type:     c.Type,
+		Count:    c.Count,
+		Updated:  c.Updated,
+		Errors:   append(c.Errors),
+		Warnings: append(c.Warnings),
+	}
+}
+
+// Coverage This is a placeholder struct which can be replaced by a more specific implementation later
+type CoverageSet struct {
+	Window Window               `json:"window"`
```

Comment:
Why do both `Coverage` and `CoverageSet` have windows? It feels to me like the Window of a `CoverageSet` should be derived (e.g. `.Window()`) from the stored Coverages' Windows. Is that wrong?

Reply:
@michaelmdresser I think that deriving that logic from the stored coverage windows would be annoying to calculate every time you wanted the range if your coverages are stored in an unsorted list. It's likely easier to modify the window of the set as new coverages are added (via the `Expand` method). 

However, you _could_ store coverages in order based on their windows, and be able to quickly access `coverage[0].Start -> coverage[len].End`